sdk/core/Azure.Core/samples/Serialization.md

```diff
@@ -1,34 +1,47 @@
-# Azure.Core public serialization samples
+# Azure.Core Public Serialization Guide
 
-## Using explicit cast
+The latest updates to the `Azure.Core` library have simplified serializing and deserializing of all public Azure models. With the addiion of the `ModelSerializer` class, we provide a public interface that exposes the currently internal serialization code so customers can access it without needing to use reflection or write their own translation. The recommended serialization approach is to use the static `ModelSerializer` class. The `ModelSerializer` class also has the `ModelSerializerOptions` which allows the user to set the serialization `ModelSerializerFormat` type (`Json` or `Wire`) and provide custom Serializer types for specific models.
```

Comment:
KrzysztofCwalina 

This doc should not start with this section. It's impossible to understand it without the context described later in this document.

annelo-msft 

It might be helpful to begin with a short summary of what the interfaces are and how they are intended to be used. For Azure.Core, it can be helpful to think about this from the perspective of an end-user of a library, as well as a library author. Should library authors implement IModelSerializable on all models? Will the generator handle it for them? When do they need to also implement IModelJsonSerializable? What does an end-user need to know they can do with a model that implements these interfaces? Should they understand ModelSerializer and how to use it? When do I need to be thinking about ModelJsonConverter and why? It'd be great to get all these questions answered at the start with some short illustrative samples before diving into further details and all the options types below. That way, when I get to those sections, as a reader I would know what I'm looking at and why it's useful to me.